server/src/test/java/io/crate/analyze/AlterTableAddColumnAnalyzerTest.java

```diff
@@ -278,6 +280,39 @@ public void testAddNewNestedObjectColumn() throws Exception {
         assertThat((String) y.get("type"), is("keyword"));
     }
 
+    @Test
+    public void test_check_constraint_can_be_added_to_nested_object_() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("""
+                create table t (
+                    i int,
+                    o object
+                )
+            """)
+            .build();
+
+        // non-existing object, used to throw ColumnUnknownException
+        BoundAddColumn analysis = analyze("alter table t add column o2['a']['b'] text " +
+            "constraint nested_col_check check (o2['a']['b']!='test')");
+        Map<String, Object> meta = (Map<String, Object>) analysis.mapping().get("_meta");
+        Map<String, String> checks = (Map<String, String>) meta.get("check_constraints");
+        assertThat(checks.get("nested_col_check")).isEqualTo("\"o2\"['a']['b'] <> 'test'");
+
+        // existing object, used to throw error "CHECK expressions defined in this context cannot refer to other columns"
+        analysis = analyze("alter table t add column o['a']['b'] text " +
+            "constraint nested_col_check check (o['a']['b']!='test')");
+        meta = (Map<String, Object>) analysis.mapping().get("_meta");
+        checks = (Map<String, String>) meta.get("check_constraints");
+        assertThat(checks.get("nested_col_check")).isEqualTo("\"o\"['a']['b'] <> 'test'");
+
+        // adding object with multiple children
+        analysis = analyze("alter table t add column o3 object as (a object as (b text), c text) " +
+            "constraint nested_col_check check (o3['a']['b']!='test')");
+        meta = (Map<String, Object>) analysis.mapping().get("_meta");
+        checks = (Map<String, String>) meta.get("check_constraints");
+        assertThat(checks.get("nested_col_check")).isEqualTo("\"o3\"['a']['b'] <> 'test'");
```

Comment:
Can we split this into dedicated tests?

Reply:
doen in https://github.com/crate/crate/pull/12938/commits/f1e93bc50d1105a1e636f68f8aa410a4f145a1d8, also update changes entry to mention that it's about sub column.